lib/tenios/api/client.rb

```diff
@@ -24,10 +22,22 @@ def http_client
         end
       end
 
-      def call_detail_records
-        @call_detail_records ||= CallDetailRecords.new(self)
+      class <<self
+        private
+
+        def endpoint(name, klass)
+          API.autoload klass, "tenios/api/#{name}"
+
+          class_eval <<~RUBY, __FILE__, __LINE__ + 1
+            def #{name}
+              @#{name} ||= #{klass}.new(self)
+            end
+          RUBY
+        end
       end
-      alias cdrs call_detail_records
+
+      endpoint :call_detail_records, :CallDetailRecords
+      endpoint :verification, :Verification
```

Comment:
A bit of meta programming to make it fun 🙃
The idea of using it here (other than DRYing) is to enforce consistency on the API client structure. Each (set of) endpoints should have its own class which follow the same pattern.